packages/esm-service-queues-app/src/create-queue-entry/queue-fields/queue-fields.component.tsx

```diff
@@ -87,76 +74,117 @@ const QueueFields = React.memo(({ setOnSubmit, defaultInitialServiceQueue }: Que
 
   const { queues, isLoading: isLoadingQueues } = useQueues(queueLocation);
   const memoizedQueues = useMemo(
-    () => queues,
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [JSON.stringify(queues.map((q) => ({ uuid: q.uuid, name: q.name })))],
+    () => queues.map((q) => ({ uuid: q.uuid, name: q.name, allowedPriorities: q.allowedPriorities })),
+    [queues],
   );
-  const priorities = queues.find((q) => q.uuid === queueService)?.allowedPriorities ?? [];
+  const priorities = useMemo(() => {
+    return memoizedQueues.find((q) => q.uuid === queueService)?.allowedPriorities ?? [];
+  }, [memoizedQueues, queueService]);
+
   const sortWeight = priority === emergencyPriorityConceptUuid ? 1 : 0;
+  const isDataLoaded = !isLoadingQueueLocations && !isLoadingQueues;
+
+  const hasCompleteQueueEntry = queueLocation && queueService && priority;
 
   const onSubmit = useCallback(
     async (visit: Visit) => {
-      const isValid = await trigger(['queueLocation', 'queueService', 'priority']);
-
-      if (!isValid) {
-        const errorMessages = Object.entries(errors)
-          .map(([, error]) => `${error?.message ?? 'Invalid'}`)
-          .join('\n');
-        return Promise.reject(new Error(`Form validation failed:\n${errorMessages}`));
-      }
-      return postQueueEntry(
-        visit.uuid,
-        queueService,
-        visit.patient.uuid,
-        priority,
-        defaultStatusConceptUuid,
-        sortWeight,
-        queueLocation,
-        visitQueueNumberAttributeUuid,
-      )
-        .then(() => {
-          showSnackbar({
-            kind: 'success',
-            isLowContrast: true,
-            title: t('addedPatientToQueue', 'Added patient to queue'),
-            subtitle: t('queueEntryAddedSuccessfully', 'Queue entry added successfully'),
-          });
-          mutateQueueEntries();
-        })
-        .catch((error) => {
-          // Check if this is a duplicate patient error with fallback handling
-          const errorMessage = error?.responseBody?.error?.message || error?.message || '';
-          const isDuplicatePatientError = errorMessage.includes(DUPLICATE_QUEUE_ENTRY_ERROR_CODE);
+      try {
+        if (!hasCompleteQueueEntry) {
+          if (!queueLocation) {
+            return Promise.resolve();
+          }
 
-          if (isDuplicatePatientError) {
```

Comment:
Any reason why we're removing the specific handling of duplicate queue entries? This regresses the UX by showing generic error messages instead of more specific ones like the helpful 'Patient already in queue' error message for duplicate entries.